plugins/modules/database/influxdb/influxdb_retention_policy.py

```diff
@@ -115,6 +129,51 @@
 from ansible.module_utils._text import to_native
 
 
+VALID_DURATION_REGEX = re.compile(r'^(\d+(ns|u|µ|ms|s|m|h|d|w))+$')
+
+DURATION_REGEX = re.compile(r'(\d+)(ns|u|µ|ms|s|m|h|d|w)')
+EXTENDED_DURATION_REGEX = re.compile(r'(?:(\d+)(ns|u|µ|ms|m|h|d|w)|(\d+(?:\.\d+)?)(s))')
+
+
+def check_duration_literal(value):
+    return VALID_DURATION_REGEX.search(value) is not None
+
+
+def parse_duration_literal(value, extended=False):
+    duration = 0.0
+
+    if value == "INF":
+        return duration
+
+    lookup = (EXTENDED_DURATION_REGEX if extended else DURATION_REGEX).findall(value)
+
+    for duration_literal in lookup:
+        if extended and duration_literal[3] == 's':
+            duration_val = float(duration_literal[2])
+            duration += duration_val * 1000 * 1000 * 1000
+        else:
+            duration_val = int(duration_literal[0])
+
+            if duration_literal[1] == 'ns':
+                duration += duration_val
+            elif duration_literal[1] == 'u' or duration_literal[1] == 'µ':
+                duration += duration_val * 1000
+            elif duration_literal[1] == 'ms':
+                duration += duration_val * 1000 * 1000
+            elif duration_literal[1] == 's':
+                duration += duration_val * 1000 * 1000 * 1000
+            elif duration_literal[1] == 'm':
+                duration += duration_val * 1000 * 1000 * 1000 * 60
+            elif duration_literal[1] == 'h':
+                duration += duration_val * 1000 * 1000 * 1000 * 60 * 60
+            elif duration_literal[1] == 'd':
+                duration += duration_val * 1000 * 1000 * 1000 * 60 * 60 * 24
+            elif duration_literal[1] == 'w':
+                duration += duration_val * 1000 * 1000 * 1000 * 60 * 60 * 24 * 7
```

Comment:
Not a *problem*, but as a suggestion: instead of this long chain of `if`s, another simple (and allegedly faster) way of doing this could be:
```suggestion
# at the beginning of the file:
DURATION_MULTIPLIERS = {
    'ns': 1,
    'u': 1000, 'µ': 1000,
    'ms': 1000 * 1000,
    's': 1000 * 1000 * 1000,
    'm': 1000 * 1000 * 1000 * 60,
    'h': 1000 * 1000 * 1000 * 60 * 60,
    'd': 1000 * 1000 * 1000 * 60 * 60 * 24,
    'w': 1000 * 1000 * 1000 * 60 * 60 * 24 * 7,
}

# ... then back to this part of the code:
            duration += duration_val * DURATION_MULTIPLIERS[duration_literal[1]]
```
of course this could also be used in line 153 above.